hdf5-sys/README.md

```diff
@@ -0,0 +1,32 @@
+# `hdf5-sys`
+
+This crate provides direct bindings to the HDF5 C library and allows to build
+the library from C sources if need be, so it can be linked in statically.
+
+### Static linking
+
+This crate supports linking to a static build of HDF5. The HDF5 C library is built
+via the `hdf5-src` crate which is then linked in when the `static` feature is set.
+See below for a list of supported options for static builds.
+
+As of the time of writing, the version of the HDF5 library that is built is 1.10.5,
+but it may be incremented later.
+
+## Crate features
+
+Features propagated upwards will be detected based on features available for the chosen library. The library selected can always be overridden by specifying the environment variable `HDF5_DIR`.
+
+General features:
+
+- `mpio`: enable MPI support (in static mode, also enable it in the HDF5 C library)
+- `static`: build HDF5 C library from source (see below for a list of options).
+
+The following features affect the HDF5 build options when compiling it from source:
+
+- `mpio`: enable MPI support
```

Comment:
@magnusuMET Should this be removed as we no longer support mpio/static together?

Reply:
I added a note regarding mutual exclusivity in `hdf5-sys/README.md`. We could add a note there that this restriction could be lifted.

Reply:
@magnusuMET Yea, but "mpio" version is no longer supported here technically, no (as of this moment)?

Could add a separate note below that we don't support mpio for static build but it could be lifted/implemented if really needed.

Reply:
`mpio` is supported for non-static builds, this feature was there before this PR. I don't think we have CI for this.

Reply:
@magnusuMET That's not what I mean :) Currently we have:

```markdown
The following features affect the HDF5 build options when compiling it from source:

- `mpio`: enable MPI support
```

Which is not true because we don't support mpio when compiling it from source? (the `mpio` notice above as a "general feature" makes sense of course, we don't test it on CI but it works for normal dynamic builds)

Reply:
Now I see what you mean! Thanks for the patience :+1:

Reply:
@aldanor Forgot to mention you..